### PR TITLE
Blogpost for Python on the Playground

### PR DESCRIPTION
**Pull request by SferaDev**

### CODE REVIEW

File: playground-python.mdx [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+The next step was to add support for Python. We wanted to have the same experience as with TypeScript and SQL, where you could write code and execute it in the browser. Luckily, there are several projects that allow you to run Python in the browser, and we decided to use [Pyodide](https://pyodide.org).
+
+Pyodide is a project started by Mozilla that allows you to run Python in the browser using WebAssembly. It includes the standard library and several popular packages like NumPy, Pandas, or Matplotlib. It also includes a package manager that allows you to install any other package with a wheel available on PyPI.
+
+Similarly to TypeScript, we have a web worker that executes the Python code. The code is executed by Pyodide, and we load external packages from imports using the package manager.
+
+To make it fully work with our Python SDK, we only needed patch the runtime so that Pyodide could do HTTP requests, using `pyodide-http`.
+
+## Conclusion
+
```
Comment (SferaDev) [inline playground-python.mdx:61]: Maybe another picture with a complex python program now?

File: playground-python.mdx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+At Xata, we're committed to providing you with a versatile and powerful UI to interact with your data. That's why early on we built the [Xata Playground](https://xata.io/playground), a web-based IDE that allows you to write code in TypeScript and SQL to query your data in Xata.
+
+Today, we're excited to announce that we've expanded the Xata Playground to include Python. Now you can write code in Xata's Playground using Python, TypeScript, and SQL.
+
+In this post, we'll do a technical deep dive into how the Xata Playground works and how we added Python to the mix.
+
+## The Xata Playground
+
+As a new hire that had just marked my first month at Xata, the 8th of April 2022, I opened a pull request with the first version of the Xata Playground. It was a very simple proof of concept that I wanted to show during our weekly team meeting.
```
Comment (alexfrancoeur) [inline playground-python.mdx:21]:
```suggestion
As a new hire that had just marked my first month at Xata, I opened a pull request with the first version of the Xata Playground. It was a very simple proof of concept that I wanted to show during our weekly team meeting.
```

I don't know if the date is necessary

File: playground-python.mdx [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+It was a simple [monaco-editor](https://microsoft.github.io/monaco-editor/) where you could write TypeScript code and run it to see the results in a separate panel. The main goal was that our users could try out our TypeScript SDK without having to install anything.
```
Comment (alexfrancoeur) [inline playground-python.mdx:23]:
```suggestion
The pull request introduced a [monaco-editor](https://microsoft.github.io/monaco-editor/) where you could write TypeScript code and run it to see the results in a separate panel. The main goal was that our users could try out our TypeScript SDK without having to install anything.
```

"It was a simple" is repeated in two back to back sentences.

File: playground-python.mdx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+Both internally and with some early adopters, we noticed that it was something we had to invest in, and we decided to iterate quickly over the proof of concept and release it as soon as possible. On the 2nd of May 2022, we released the first version of the Xata Playground.
```
Comment (alexfrancoeur) [inline playground-python.mdx:25]:
```suggestion
Both internally and with some early adopters, we noticed that it was something we had to invest in, and decided to iterate quickly over the proof of concept and release it as soon as possible. A month later, we released the first version of the Xata Playground. And it's still one of our most popular features.
```

The exact date felt a bit unnecessary and I was thinking we could call out how popular it is.

File: playground-python.mdx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+<ArticleImage
+  src="https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/playground-v1.png"
```
Comment (alexfrancoeur) [inline playground-python.mdx:28]: What an image! Wow, Xata has come a long way

File: playground-python.mdx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+
+Also, to improve the experience when writing TypeScript code, we switched to the fork of monaco-editor that [TypeScript Playground](https://www.typescriptlang.org/play) uses. This fork is always up to date with the latest version of TypeScript, and it also includes several improvements for TypeScript developers.
+
+## Adding support for multiple files and languages
+
+The first version of the Playground only supported a single TypeScript file. This was a limitation that we wanted to remove, and when we started working on it, we also decided to add support for other languages.
+
+Monaco is the open source editor that powers VS Code, and it has support for multiple files and languages. We just needed to use its virtual file system that would allow us to load multiple files and execute them separately.
+
+The refactor allowed us to add support for SQL, both for full file execution and for inline execution. This was a great addition, as it allowed us to show how Xata's SQL Proxy works, and how you can use it to query your data.
```
Comment (alexfrancoeur) [inline playground-python.mdx:51]:
```suggestion
The refactor allowed us to add support for SQL, both for full file execution and for inline execution. This was a great addition, as it allowed us query your data with [SQL over HTTP](https://xata.io/blog/sql-over-http).
```

I don't know if this is any better, but SQL proxy is a bit of an internal term

File: playground-python.mdx (hunk 1 above)
Comment (alexfrancoeur) [reply playground-python.mdx:64]: Yeah, I was going to suggest the same. Right after we introduce Python, let's show the new UI with a few saved queries and Python support

**Review by alexfrancoeur — APPROVED** [review]
This is great @SferaDev! I dropped in a few suggestions and comments, but they're minor.  Looking forward to releasing Python support!

File: playground-python.mdx [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/playground-v1.png
+  alt: Xata Playground
+author: Alexis Rico
+date: 09-19-2023
+published: false
+slug: playground-python
+---
+
+At Xata, we're committed to providing you with a versatile and powerful UI to interact with your data. That's why early on we built the [Xata Playground](https://xata.io/playground), a web-based IDE that allows you to write code in TypeScript and SQL to query your data in Xata.
```
Comment (SferaDev) [inline playground-python.mdx:13]:
```suggestion
At Xata, we're committed to providing you with a versatile and powerful UI to interact with your data. That's why early on we built the Xata Playground, a web-based IDE that allows you to write code in TypeScript and SQL to query your data in Xata.
```

File: playground-python.mdx [hunk 8]
```diff
@@ -0,0 +1,67 @@
+---
+title: 'Expanding the Xata Playground: Python Joins TypeScript and SQL'
+description: 'We have expanded the Xata Playground to include Python. Now you can write code in Xata Playground using Python, TypeScript, and SQL.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/playground-v1.png
```
Comment (SferaDev) [inline playground-python.mdx:5]:
```suggestion
  src: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/xata-python-logos.png
```

File: playground-python.mdx [hunk 9]
```diff
@@ -0,0 +1,85 @@
+---
+title: 'Xata Playground now runs python in the browser'
+description: 'We have expanded the Xata Playground to include Python. Now you can write code in Xata Playground using Python, TypeScript, and SQL.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/xata-python-logos.png
```
Comment (SferaDev) [inline playground-python.mdx:5]:
```suggestion
  src: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/xata-python-logos.png
```

File: playground-python.mdx [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+## The Xata Playground
+
+As a new hire that had just marked my first month at Xata, I opened a pull request with the first version of the Xata Playground. It was a very simple proof of concept that I wanted to show during our weekly team meeting.
+
+The pull request introduced a [monaco-editor](https://microsoft.github.io/monaco-editor/) where you could write TypeScript code and run it to see the results in a separate panel. The main goal was that our users could try out our TypeScript SDK without having to install anything.
+
+Both internally and with some early adopters, we noticed that it was something we had to invest in, and decided to iterate quickly over the proof of concept and release it as soon as possible. A month later, we released the first version of the Xata Playground. And it's still one of our most popular features.
+
+<ArticleImage
+  src="https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/playground-v1.png"
```
Comment (SferaDev) [inline playground-python.mdx:28]:
```suggestion
  src="https://raw.githubusercontent.com/xataio/mdx-blog/main/images/playground-v1.png"
```

File: playground-python.mdx [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+To achieve this, we use rollup to bundle the code and a [plugin](https://github.com/SferaDev/rollup-plugin-import-cdn) to load any external library from a CDN. This way, we can transpile the code and all its dependencies into a single file that can be executed in a separate thread using [Web Workers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Using_web_workers).
+
+Running the code in a separate thread is important, as it allows us to avoid blocking the main thread. This way, the UI is always responsive, and you can continue writing code while the previous code is being executed. Also, it provides some built-in security, as the code is isolated from the main thread.
+
+Also, to improve the experience when writing TypeScript code, we switched to the fork of monaco-editor that [TypeScript Playground](https://www.typescriptlang.org/play) uses. This fork is always up to date with the latest version of TypeScript, and it also includes several improvements for TypeScript developers.
+
+<ArticleImage
+  src="https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/playground-v2.png"
```
Comment (SferaDev) [inline playground-python.mdx:48]:
```suggestion
  src="https://raw.githubusercontent.com/xataio/mdx-blog/main/images/playground-v2.png"
```

File: playground-python.mdx [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+The next step was to add support for Python. We wanted to have the same experience as with TypeScript and SQL, where you could write code and execute it in the browser. Luckily, there are several projects that allow you to run Python in the browser, and we decided to use [Pyodide](https://pyodide.org).
+
+Pyodide is a project started by Mozilla that allows you to run Python in the browser using WebAssembly. It includes the standard library and several popular packages like NumPy, Pandas, or Matplotlib. It also includes a package manager that allows you to install any other package with a wheel available on PyPI.
+
+Similarly to TypeScript, we have a web worker that executes the Python code. The code is executed by Pyodide, and we load external packages from imports using the package manager.
+
+To make it fully work with our Python SDK, we only needed patch the runtime so that Pyodide could do HTTP requests, using `pyodide-http`.
+
+<ArticleImage
+  src="https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/playground-v3.png"
```
Comment (SferaDev) [inline playground-python.mdx:74]:
```suggestion
  src="https://raw.githubusercontent.com/xataio/mdx-blog/main/images/playground-v3.png"
```

File: playground-python.mdx [hunk 13]
```diff
@@ -0,0 +1,85 @@
+---
+title: 'Xata Playground now runs python in the browser'
+description: 'We have expanded the Xata Playground to include Python. Now you can write code in Xata Playground using Python, TypeScript, and SQL.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/xata-python-logos.png
+  alt: Xata Playground
+author: Alexis Rico
+date: 09-19-2023
+published: false
```
Comment (SferaDev) [inline playground-python.mdx:9]:
```suggestion
published: true
```

File: playground-python.mdx [hunk 14]
```diff
@@ -0,0 +1,85 @@
...
+slug: playground-python
+---
```
Comment (SferaDev) [inline playground-python.mdx:11]:
```suggestion
ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/xata-python-logos.png
---
```

File: playground-python.mdx [hunk 15]
```diff
@@ -0,0 +1,86 @@
...
+title: 'Xata Playground now runs python in the browser'
+description: 'We have expanded the Xata Playground to include Python. Now you can write code in Xata Playground using Python, TypeScript, and SQL.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/xata-python-logos.png
+  alt: Xata Playground
+author: Alexis Rico
+date: 09-19-2023
+published: false
+slug: playground-python
+ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/xata-python-logos.png
```
Comment (SferaDev) [inline playground-python.mdx:11]:
```suggestion
ogImage: https://raw.githubusercontent.com/xataio/mdx-blog/main/images/xata-python-logos.png
```

**Review by philkra — APPROVED** [review]
LGTM

File: playground-python.mdx [hunk 16]
```diff
@@ -0,0 +1,86 @@
+---
+title: 'Xata Playground now runs python in the browser'
+description: 'We have expanded the Xata Playground to include Python. Now you can write code in Xata Playground using Python, TypeScript, and SQL.'
+image:
+  src: https://raw.githubusercontent.com/xataio/mdx-blog/py-play/images/xata-python-logos.png
+  alt: Xata Playground
+author: Alexis Rico
+date: 09-19-2023
```
Comment (SferaDev) [inline playground-python.mdx:8]:
```suggestion
date: 09-18-2023
```

**Pull request merged**